test/e2e/cmd/run/run.go

```diff
@@ -478,6 +479,8 @@ func (h *helper) monitorTestJob(client *kubernetes.Clientset) error {
 	})
 
 	informer.Run(ctx.Done())
+	// wait for the log stream to be fully flushed
+	<-logStreamStopped
```

Comment:
I am curious why did you use a channel instead of a `WaitGroup` here?

Reply:
No reason. I can totally use a `WaitGroup`, it probably makes the intent clearer 👍